back/engines/commercial/analytics/app/models/analytics/fact_email_delivery.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: analytics_fact_email_deliveries
+#
+#  id                     :uuid             primary key
+#  dimension_date_sent_id :date
+#  automated              :boolean
+#  sent                   :boolean
+#
+module Analytics
+  class FactEmailDelivery < Analytics::ApplicationRecordView
+    self.primary_key = :id
+    belongs_to :dimension_date_sent, class_name: 'Analytics::DimensionDate', primary_key: 'date'
```

Comment:
```suggestion
    belongs_to :dimension_date_sent
```

It will no longer be necessary to specify the class name explicitly once #3098 is merged.

Reply:
how does that work?

Reply:
See: https://github.com/CitizenLabDotCo/citizenlab/blob/014aa1e9eca5b10b7164e3ce62f52e0ea175cc56/back/engines/commercial/analytics/lib/query.rb#L152.

Reply:
sorry, now i understand even less. What would be the association class for `dimension_date_sent` if i don't specify the class name?

Reply:
Just the same: `Analytics::DimensionDate`.

Reply:
Ah no, sorry 😑 not in that case. Forget about it :)

Reply:
ok, will remain a mystery then, will ask you again when i build another data model.